main/src/ca/uwaterloo/flix/language/phase/unification/CaseSetUnification.scala

```diff
@@ -49,25 +50,35 @@ object CaseSetUnification {
       return Ok(Substitution.empty)
     }
 
-//        tpe1 match {
-//          case x: Type.Var if renv.isFlexible(x.sym) =>
-//            if (tpe2 eq Type.All)
-//              return Ok(Substitution.singleton(x.sym, Type.All))
-//            if (tpe2 eq Type.Empty)
-//              return Ok(Substitution.singleton(x.sym, Type.Empty))
-//
-//          case _ => // nop
+    // TODO RESTR-VARS this doesn't help lol
```

Comment:
It does not help? :(

Reply:
Makes it worse somehow. But this was a naive translation from the other file. Probably something smarter can be done here.